gazebo/rendering/RTShaderSystem.cc

```diff
@@ -636,20 +593,33 @@ Ogre::PSSMShadowCameraSetup *RTShaderSystem::GetPSSMShadowCameraSetup() const
 void RTShaderSystem::Update()
 {
   GZ_PROFILE("rendering::RTShaderSystem::Update");
-  if (!this->dataPtr->initialized || !this->dataPtr->updateShaders)
+  if (!this->dataPtr->initialized)
   {
     return;
   }
 
-  for (const auto &scene : this->dataPtr->scenes)
+  std::lock_guard<std::mutex> lock(this->dataPtr->updateMutex);
```

Comment:
nit: if we want to minimize the time spent in this lock, we could only lock it while copying the values of `this->dataPtr->updateShadows` and `this->dataPtr->updateShaders`, then reset both to `false`, and then release the mutex

for example:

~~~
bool updateShaders, updateShadows;
{
  std::lock_guard<std::mutex> lock(this->dataPtr->updateMutex);
  updateShaders = this->dataPtr->updateShaders;
  updateShadows = this->dataPtr->updateShadows;
  this->dataPtr->updateShaders = false;
  this->dataPtr->updateShadows = false;
}
~~~

then read from the local variable instead of the member variable

Reply:
yep done. 5f426b2